papirus-folders: fix missing dependency

aacebedo opened the pull request:
###### Description of changes

The getent command was missing when the package was used in papirus-icon-theme.

Added it as a propagatedBuildInputs

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

aacebedo committed:
papirus-folders: fix missing dependency

The getent command was missing when the package was used in papirus-icon-theme.

Added it as a propagatedBuildInputs

aacebedo commented:
@romildo By any chance would you have some time to review this please ?
It is linked to the color argument in `papirus-icon-theme` and I don't know why the bug didn't show up when I tested it.
I added a `propagatedBuildInputs` for `getent` hoping it will be available when installing `papirus-icon-theme`

romildo commented:
> @romildo By any chance would you have some time to review this please ? It is linked to the color argument in `papirus-icon-theme` and I don't know why the bug didn't show up when I tested it. I added a `propagatedBuildInputs` for `getent` hoping it will be available when installing `papirus-icon-theme`

When I have first tested building `papirus-icon-theme` with a different color it worked for me. I have used the command line:
```console
$ nix-build --option sandbox true -K -E 'with import ./. {}; papirus-icon-theme.override { color="red"; }'
```
But now it is failing too.

On my NixOS system `getent` is already found in `$PATH`. It is provided by `glibc.bin`:
```console
$ readlink -f $(which getent)
/nix/store/slkw2qamzs6w441jgan1nj64xzwcgwmw-glibc-2.34-210-bin/bin/getent
```
I do not know why it is not found when building `papirus-icon-theme` anymore. 

Adding `getent` as a (propagated or not) dependence will not add `${getent}/bin` into `$PATH` when `papirus-folders` is installed. It would be available in `$PATH` only at build time.

There are two solutions for the problem here:
1. wrapp the `papirus-folders` script in order to add `${getent}/bin` to the `PATH` environment variable before calling the original script
2. patch the original script `papirus-folders` to explicitly include the full path to `gentent`

I prefer the second solution.

You can just add something like the following (untested):

```nix
  postPatch = ''
    substituteInPlace papirus-folder --replace getent ${getent}/bin/getent
  ''
```

aacebedo commented:
> > @romildo By any chance would you have some time to review this please ? It is linked to the color argument in `papirus-icon-theme` and I don't know why the bug didn't show up when I tested it. I added a `propagatedBuildInputs` for `getent` hoping it will be available when installing `papirus-icon-theme`
> 
> When I have first tested building `papirus-icon-theme` with a different color it worked for me. I have used the command line:
> 
> ```
> $ nix-build --option sandbox true -K -E 'with import ./. {}; papirus-icon-theme.override { color="red"; }'
> ```
> 
> But now it is failing too.
> 
> On my NixOS system `getent` is already found in `$PATH`. It is provided by `glibc.bin`:
> 
> ```
> $ readlink -f $(which getent)
> /nix/store/slkw2qamzs6w441jgan1nj64xzwcgwmw-glibc-2.34-210-bin/bin/getent
> ```
> 
> I do not know why it is not found when building `papirus-icon-theme` anymore.
> 
> Adding `getent` as a (propagated or not) dependence will not add `${getent}/bin` into `$PATH` when `papirus-folders` is installed. It would be available in `$PATH` only at build time.
> 
> There are two solutions for the problem here:
> 
>     1. wrapp the `papirus-folders` script in order to add `${getent}/bin` to the `PATH` environment variable before calling the original script
> 
>     2. patch the original script `papirus-folders` to explicitly include the full path to `gentent`
> 
> 
> I prefer the second solution.
> 
> You can just add something like the following (untested):
> 
> ```nix
>   postPatch = ''
>     substituteInPlace papirus-folder --replace getent ${getent}/bin/getent
>   ''
> ```

Got it, agree with you for the second solution. Modified it and tested it.
